q2_moshpit/busco/utils.py

```diff
@@ -44,461 +49,163 @@ def _parse_busco_params(arg_key, arg_val) -> List[str]:
-                "category",
-                scale=alt.Scale(domain=domain, range=range_),
-                legend=alt.Legend(title="BUSCO Category", orient="top"),
-            ),
-            order=alt.Order("order", sort="ascending"),
-            tooltip=[
-                alt.Tooltip("sample_id", title="Sample ID"),
-                alt.Tooltip("mag_id", title="MAG ID"),
-                alt.Tooltip("dataset", title="Lineage dataset"),
-                alt.Tooltip(
-                    "fracc_markers",
-                    title="Aprox. number of markers in this category"
-                ),
-                alt.Tooltip("BUSCO_percentage", title="% BUSCOs"),
-            ],
-            opacity=alt.value(0.85),
-        )
-        .properties(
-            width=width,
-            height={"step": height}
-        )
-        .facet(
-            row=alt.Row(
-                "sample_id",
-                title="Sample ID / MAG ID"
-            ),
-            spacing=spacing
-        )
-        .resolve_scale(y="independent")
-    )
-
-    # Secondary plot
-    # Drop down menu
-    dropdown = alt.binding_select(
-        options=[
-            'scaffold_n50',
-            'contigs_n50',
-            'percent_gaps',
-            'number_of_scaffolds',
-        ],
-        name="Assembly Statistics: "
-    )
-
-    xcol_param = alt.param(
-        value='scaffold_n50',
-        bind=dropdown
-    )
-
-    secondary_plot = alt.Chart(secondary_plot_data).mark_bar().encode(
-        x=alt.X('x:Q').title('Assembly Statistic'),
-        y=alt.Y('mag_id', axis=None),
-        tooltip=[alt.Tooltip('x:Q', title="value")],
-        opacity=alt.value(0.85)
-    ).transform_calculate(
-        x=f'datum[{xcol_param.name}]'
-    ).add_params(
-        xcol_param
-    ).properties(
-        width=width,
-        height={"step": height}
-    ).facet(
-        row=alt.Row(
-            "sample_id",
-            title=None,
-            header=alt.Header(labelFontSize=0),
-        ),
-        spacing=spacing
-    ).resolve_scale(
-        y="independent"
-    )
-
-    # concatenate plots horizontally
-    output_plot = alt.hconcat(
-        busco_plot, secondary_plot, spacing=3
-    ).configure_axis(
-        labelFontSize=labelFontSize, titleFontSize=titleFontSize
-    ).configure_legend(
-        labelFontSize=labelFontSize, titleFontSize=titleFontSize
-    ).configure_header(
-        labelFontSize=labelFontSize, titleFontSize=titleFontSize
-    )
-
-    # Return
-    return output_plot.to_json()
-
-
-def _run_busco(
-    output_dir: str, mags: MultiMAGSequencesDirFmt, params: List[str]
-) -> Dict[str, str]:
-    """Evaluates bins for all samples using BUSCO.
+    This function groups the DataFrame by 'sample_id' and then partitions
+    these groups into smaller DataFrames. Each partition will have a total
+    row count less than or equal to the max_rows parameter. The last group
+    in a partition can exceed the max_rows limit.
```

Comment:
```suggestion
    This function groups the DataFrame by 'sample_id' and then partitions
    these groups into smaller DataFrames. Provided that no sample has 
    more than 'max_rows' MAGs (i.e. rows), each partition will have a total
    row count less than or equal to the 'max_rows'.
```

I think the above description is more appropriate since the first partition can also exceed the limit if, for example, the first sample has 101 MAGs/rows.